test/arguments.ts

```diff
@@ -375,8 +375,23 @@ test('`context` option is accessible when extending instances', t => {
 
 	const instance = got.extend({context});
 
-	t.is(instance.defaults.options.context, context);
-	t.false({}.propertyIsEnumerable.call(instance.defaults.options, 'context'));
+	t.deepEqual(instance.defaults.options.context, context);
+	t.true({}.propertyIsEnumerable.call(instance.defaults.options, 'context'));
+});
+
+test('`context` option is shallow merged', t => {
+	const context = {
+		foo: 'bar'
+	};
+
+	const instance1 = got.extend({context});
+
+	t.deepEqual(instance1.defaults.options.context, context);
+	t.true({}.propertyIsEnumerable.call(instance1.defaults.options, 'context'));
+
+	const instance2 = instance1.extend({context: {}});
```

Comment:
Maybe put some properties into the second context.
This test is going to pass even if the second context is ignored.

Reply:
done!